### PR TITLE
Feature: Clan focus

### PR DESCRIPTION
**Pull request by Lixxis**
New feature which allows to influence the general direction of the clan / how the warriors "behave"
This allows following interactions:
* managing roughly resources (prey & herb)
* lowering (or increase) the chance of injuries/illnesses 
* influence the reputation for other clans + outsiders

There are still some smaller bugs which are tracked in the according thread in discord.

### CODE REVIEW

File: resources/game_config.json [hunk 1]
```diff
@@ -294,6 +294,102 @@
...
+            ]
+        },
+        "rest and recover": {
+            "injury_prevent": 4,
+            "illness_prevent": 6,
+            "outbreak_prevention": 2,
+			"moons_earlier_healed": 1,
+            "comments": [
+                "injury will be roughly prevented for every fourth cat",
+                "illness will be roughly prevented for every sixt cat - illness is not as easy preventable as injuries",
```
Comment (Thlumyn) [inline resources/game_config.json]: *sixth

File: resources/game_config.json [hunk 2]
```diff
@@ -294,6 +294,102 @@
...
+            "med_apprentice" : 2,
+            "comment": "Defines how much a working warrior(leader/deputy) or (apprentice) medicine cat gathers additionally."
+        },
+        "outsiders" : {
+            "reputation": 5,
+            "comment": "Defines how much increase or decrease the reputation is."
+        },
+        "other clans" : {
+            "relation": 2,
+            "comment": "Defines how much increase or decrease the relation with other clans."
```
Comment (Thlumyn) [inline resources/game_config.json]: *with other clans is.

**Review by Thlumyn — CHANGES_REQUESTED** [review]
A couple things, aside from the spelling issues that I noted:
Some of the focus icons have a background, and some do not? 
Once you are done testing the feature, the print statement about rest and recovery should be removed in the future; I think we're getting a few too many print statements building up

**Comment by Lixxis** [comment]
@Thlumyn What do you mean with background? All focus types have in the spelling bubble an own picture - or did I miss something?

I also removed or comment the prints out for the merge. The feature was already tested on my branch for several weeks. I want to have the feature on the development branch now, so I don't have to worry about merge conflicts anymore.

File: scripts/events.py [hunk 3]
```diff
@@ -578,6 +579,250 @@ def herb_destruction(self):
...
+                Cat.all_cats.values()
+            ))
+            app_amount = len(healthy_apprentices) * game.config["focus"]["hunting"]["apprentice"]
+            
+            # finish
+            total_amount = warrior_amount + app_amount
+            game.clan.freshkill_pile.add_freshkill(total_amount)
+            if total_amount > 1:
+                focus_text = f"With the heightened focus of the Clan, {total_amount} additional pieces of prey were gathered."
+            if total_amount == 1:
```
Comment (sedgestripe) [inline scripts/events.py]: should be an elif so this doesn't get overwritten in the else block

File: scripts/events.py [hunk 4]
```diff
@@ -578,6 +579,250 @@ def herb_destruction(self):
...
+            idx = 0
+            for herb, amount in herb_counter.items():
+                log_text += str(amount) + " " + herb.replace("_", " ")
+                idx += 1
+                if idx < len(herb_counter) - 1:
+                    log_text += ", "
+                elif idx < len(herb_counter):
+                    log_text += " and "
+            log_text += "."
+            game.herb_events_list.extend(log_text)
```
Comment (sedgestripe) [inline scripts/events.py]: append instead of extend since log_text is a string?

**Review by sedgestripe — CHANGES_REQUESTED** [review]

**Comment by Thlumyn** [comment]
> @Thlumyn What do you mean with background? All focus types have in the spelling bubble an own picture - or did I miss something?


image

Like, how some of them have the background image baked in; I tried it on both light mode and dark mode and it doesn't really make a visual difference, but is there a reason why some are like that? If not, it'd be nice for consistency if they were all the same

**Comment by scribblecrumb** [comment]
> > @Thlumyn What do you mean with background? All focus types have in the spelling bubble an own picture - or did I miss something?
> 
> image
> 
> Like, how some of them have the background image baked in; I tried it on both light mode and dark mode and it doesn't really make a visual difference, but is there a reason why some are like that? If not, it'd be nice for consistency if they were all the same

Hmm, interesting.  It's been so long since I made the graphics I honestly don't remember if there was a reason.  At this point I'd just guess that it was a mistake on my part, I probably had to make some edits at some point and saved the .pngs with the base layer visible by accident.

**Review by sedgestripe — APPROVED** [review]

**Comment by Lixxis** [comment]
@Thlumyn is there still something I might have missed / did you found any errors?

**Review by Ryos00 — APPROVED** [review]

**Review by Thlumyn — APPROVED** [review]

**Pull request merged**